src/tasks/generateSignatures.ts

```diff
@@ -0,0 +1,102 @@
+import fs from "fs";
+
+import "hardhat-deploy";
+import "@nomiclabs/hardhat-ethers";
+import { task } from "hardhat/config";
+
+import { TypedDataDomain } from "../../src/ts/ethers";
+
+export function domain(
+  chainId: number,
+  verifyingContract: string,
+): TypedDataDomain {
+  return {
+    name: "AccessManager",
+    version: "v1",
+    chainId,
+    verifyingContract,
+  };
+}
+import { getAllowListOffChainManagedContract } from "./utils";
+
+// interface Signature {
+//   address: string;
+//   signature: string;
+// }
+
+// interface SignatureSet {
+//   auctionId: number;
+//   chainId: number;
+//   signatures: Signature;
+// }
+
+const generateSignatures: () => void = () => {
+  task(
+    "generateSignatures",
+    "Generates the signatures for the allowListManager",
+  )
+    .addParam("auctionId", "Id of the auction ")
+    .addParam(
+      "fileWithAddress",
+      "File with comma separated addresses that should be allow-listed",
+    )
+    .setAction(async (taskArgs, hardhatRuntime) => {
+      const [caller] = await hardhatRuntime.ethers.getSigners();
+      console.log(
+        "Using the account: ",
+        caller.address,
+        " to generate signatures",
+      );
```

Comment:
I think it would be nice to check that `caller.address` is indeed used for signature verification for this auction ID by reading the actual verified from the settlement contract and comparing.